File: DigiForum_BE/DigiForum_BE/Models/Post.cs
﻿namespace DigiForum_BE.Models
{
    public record Post
    {
        public Guid PostId { get; set; } = Guid.NewGuid();
        public Guid UserId { get; set; } 
        public string Title { get; set; }
        public string Content { get; set; }
        public string ImagePath { get; set; } 
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
        public string Status { get; set; }

        public User? User { get; set; }
    }
}

File: front/src/types/Post.ts
import { User } from "./User";

export interface Post {
  key: string;
    postId: string;
    userId: string;
    title: string;
    content: string;
    imagePath: string;
    createdAt?: string;
    updatedAt?: string;
    status: string;
    user: User;
  }


export interface ReqPost {
    title : string;
content : string;
imagePath : string;
updatedAt : string;
status ?: string;
}
  